collectors/ciscomeraki/collector.js

```diff
@@ -0,0 +1,270 @@
+                    const initialStates = networks.map(networkId => ({
+                        networkId: networkId,
+                        since: startTs,
+                        until: endTs,
+                        nextPage: null,
+                        poll_interval_sec: parseInt(Math.floor(Math.random() * 30) + 1)
+                    }));
+                    return callback(null, initialStates, 1);
+                } else {
+                    return callback("Error: No networks found");
+                }
+            }
+
+        } catch (error) {
+            return callback(error);
+        }
+    }
+
+    async getAllNetworks(callback) {
+        const { clientSecret, apiEndpoint, orgKey } = this.validateAndPrepare(callback);
+        const url = `/api/v1/organizations/${orgKey}/networks`;
+        try {
+            const networks = await utils.getAllNetworks(url, clientSecret, apiEndpoint);
+            return networks.map(network => network.id);
+        } catch (error) {
+            return error;
+        }
+    }
+
+    async handleEvent(event) {
+        let collector = this;
+        let context = this._invokeContext;
+        let parsedEvent = collector._parseEvent(event);
+
+        switch (parsedEvent.RequestType) {
+            case 'ScheduledEvent':
+                switch (parsedEvent.Type) {
+                    case 'SelfUpdate':
+                        collector.handleUpdateStreamsFromNetworks();
+                        return collector.handleUpdate();
+                        break;
+                    default:
+                        super.handleEvent(event);
+                }
+            default:
+                super.handleEvent(event);
+        }
+    }
+
+    async handleUpdateStreamsFromNetworks() {
+        let collector = this;
+        //get networks from api
+        let networks = await collector.getAllNetworks();
+        const keyValue = `${process.env.customer_id}/${collector._pawsCollectorType}/${collector.collector_id}/networks_${collector.collector_id}.json`;
+
+        let params = await utils.getS3ObjectParams(keyValue, undefined);
+        //get networks json from s3 bucket
+        let networksFromS3 = await utils.fetchJsonFromS3Bucket(params.bucketName, params.key);
+        if (networks.length > 0 && Array.isArray(networksFromS3) && networksFromS3.length > 0) {
+            let differenceNetworks = utils.differenceOfNetworksArray(networks, networksFromS3);
+            AlLogger.debug(`CMRI0000024 Networks updated ${JSON.stringify(differenceNetworks)}`);
+
+            if (differenceNetworks.length > 0) {
+                const startTs = process.env.paws_collection_start_ts ?
+                    process.env.paws_collection_start_ts :
+                    moment().toISOString();
+                const endTs = moment(startTs).add(this.pollInterval, 'seconds').toISOString();
+                const initialStates = differenceNetworks.map(networkId => ({
```

Comment:
can we write common function which will set the state and called from pawsInitCollectionState and here or multiple place if it is common 
EX
` function generateInitialStates(networkIds,pollInterval ) {
    const startTs = process.env.paws_collection_start_ts ?
        process.env.paws_collection_start_ts :
        moment().toISOString();
    const endTs = moment(startTs).add(pollInterval, 'seconds').toISOString();

    const initialStates = networkIds.map(networkId => ({
        networkId: networkId,
        since: startTs,
        until: endTs,
        nextPage: null,
        poll_interval_sec: parseInt(Math.floor(Math.random() * 30) + 1)
    }));
    return initialStates;
}`

Reply:
made common method